server/routes/prisonerChangeIncentiveLevelDetails.test.ts

```diff
@@ -0,0 +1,324 @@
+  cellLocation: '123',
+}
+
+const userCaseload: UserCaseload = {
+  activeCaseload: {
+    id: 'MDI',
+    name: 'MDI',
+  },
+  caseloads: [
+    {
+      id: 'MDI',
+      name: 'MDI',
+    },
+  ],
+}
+
+const prisonApi = PrisonApi.prototype as jest.Mocked<PrisonApi>
+const offenderSearch = OffenderSearchClient.prototype as jest.Mocked<OffenderSearchClient>
+const incentivesApi = IncentivesApi.prototype as jest.Mocked<IncentivesApi>
+const nomisUserRolesApi = NomisUserRolesApi.prototype as jest.Mocked<NomisUserRolesApi>
+beforeEach(() => {
+  prisonApi.getPrisonerDetails.mockResolvedValue(prisonerDetails)
+  prisonApi.getStaffDetails.mockResolvedValue(staffDetails)
+  prisonApi.getAgency.mockImplementation(agencyId => {
+    if (agencyId === 'MDI') {
+      return Promise.resolve({
+        agencyId: 'MDI',
+        description: '123',
+        agencyType: '123',
+        active: true,
+      })
+    }
+    return Promise.resolve({
+      agencyId: 'LEI',
+      description: '123',
+      agencyType: '123',
+      active: true,
+    })
+  })
+  prisonApi.getFullDetails.mockResolvedValue(prisonerDetails)
+  offenderSearch.getPrisoner.mockResolvedValue(offenderDetails)
+  incentivesApi.getIncentiveSummaryForPrisoner.mockResolvedValue(incentiveSummaryForBooking)
+  incentivesApi.getPrisonIncentiveLevels.mockResolvedValue(samplePrisonIncentiveLevels)
+  nomisUserRolesApi.getUserCaseloads.mockResolvedValue(userCaseload)
+
+  app = appWithAllRoutes({})
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+describe('GET /incentive-reviews/prisoner/change-incentive-level', () => {
+  it('should NOT allow user to update incentive level without role', async () => {
+    return request(app)
+      .get(`/incentive-reviews/prisoner/${prisonerNumber}/change-incentive-level`)
+      .set('authorization', `bearer ${tokenWithMissingRole}`)
+      .expect(res => {
+        expect(res.redirect).toBeTruthy()
+        expect(res.headers.location).toBe('/authError')
+      })
+  })
+  it('should render the correct template with the correct data', async () => {
+    return request(app)
+      .get(`/incentive-reviews/prisoner/${prisonerNumber}/change-incentive-level`)
+      .set('authorization', `bearer ${tokenWithNecessaryRole}`)
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Smith, John')
+        expect(res.text).toContain('Record an incentive level')
+        expect(res.text).toContain('Standard (current level)')
+        expect(res.text).toContain('Reason for recording')
+        expect(res.text).toContain('Save')
+      })
+  })
+
+  it('should return 404 if prisoner is not found', () => {
+    const error: SanitisedError = {
+      name: 'Error',
+      status: 404,
+      message: 'Not Found',
+      stack: 'Not Found',
+    }
+    offenderSearch.getPrisoner.mockRejectedValue(error)
+    return request(app)
+      .get(`/incentive-reviews/prisoner/${prisonerNumber}`)
+      .expect(404)
```

Comment:
we should still include the valid authorisation bearer token so that it is purely the missing prisoner that causes this request to 404